commit 3832c115bf806baa991c51474a93909a89024525
Author: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date:   Fri Dec 21 17:45:03 2018 +0100

    arm64: dts: marvell: armada-ap806: reserve PSCI area
    
    commit 132ac39cffbcfed80ada38ef0fc6d34d95da7be6 upstream.
    
    The memory area [0x4000000-0x4200000[ is occupied by the PSCI firmware. Any
    attempt to access it from Linux leads to an immediate crash.
    
    So let's make the same memory reservation as the vendor kernel.
    
    [gregory: added as comment that this region matches the mainline U-boot]
    Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
    Signed-off-by: Gregory CLEMENT <gregory.clement@bootlin.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 176e38d54872..ec0da5b3d7fd 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -27,6 +27,23 @@
 		method = "smc";
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/*
+		 * This area matches the mapping done with a
+		 * mainline U-Boot, and should be updated by the
+		 * bootloader.
+		 */
+
+		psci-area@4000000 {
+			reg = <0x0 0x4000000 0x0 0x200000>;
+			no-map;
+		};
+	};
+
 	ap806 {
 		#address-cells = <2>;
 		#size-cells = <2>;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1057_3832c115bf806baa991c51474a93909a89024525.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1057_3832c115bf806baa991c51474a93909a89024525.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
